src/uu/mv/src/mv.rs

```diff
@@ -661,6 +661,9 @@ fn rename_with_fallback(
                 };
             }
         } else {
+            if to.is_symlink() {
+                fs::remove_file(to)?;
```

Comment:
If this fails, then the error `Operation not permitted` is very unhelpful. Can you improve upon that? (Also, please add a test that this doesn't regress in the future.)

Example on a system where the current directory is on a different partition than `/tmp`, and the current user is NOT root:

```console
$ echo "src content" > srcfile
$ echo "tmp content" > /tmp/tmpfile
$ sudo ln -s tmpfile /tmp/symlink_to_tmpfile # Create a symlink that cannot be removed
$ mv -v srcfile /tmp/symlink_to_tmpfile 
mv: inter-device move failed: 'srcfile' to '/tmp/symlink_to_tmpfile'; unable to remove target: Operation not permitted
[$? = 1]
$ cargo run -q mv -v srcfile /tmp/symlink_to_tmpfile 
mv: Operation not permitted (os error 1)
[$? = 1]
```

Reply:
thank you for guidance, I will do this